tests/mainnet/test_getters.py

```diff
@@ -0,0 +1,726 @@
+    )
+    check_pool_already_registered(metaregistry, pool, registry)
+
+    print("MetaRegistry registry_length(): ", metaregistry.registry_length())
+    print(f"Checking if pool {pool} is a metapool, in registry {registry}")
+    is_meta = metaregistry.is_meta(pool)
+    pool_coins = [coin for coin in metaregistry.get_coins(pool) if coin != brownie.ZERO_ADDRESS]
+
+    base_combinations = list(itertools.combinations(pool_coins, 2))
+    all_combinations = base_combinations
+    if is_meta:
+        underlying_coins = [
+            coin for coin in metaregistry.get_underlying_coins(pool) if coin != brownie.ZERO_ADDRESS
+        ]
+        all_combinations = all_combinations + [(pool_coins[0], coin) for coin in underlying_coins]
+
+    for combination in all_combinations:
+        if combination[0] == combination[1]:
+            continue
+        metaregistry_output = metaregistry.get_coin_indices(pool, combination[0], combination[1])
+        if registry_id > 1:
+            indices = registry.get_coin_indices(pool, combination[0], combination[1])
+            actual_output = (indices[0], indices[1], False)
+        else:
+            actual_output = registry.get_coin_indices(pool, combination[0], combination[1])
+        # fix bug with stable registry & is_underlying always true
+        if metaregistry.get_registry_handler_from_pool(pool) == stable_factory_handler.address:
+            actual_output = list(actual_output)
+            actual_output[-1] = not registry.is_meta(pool)
+        assert actual_output == metaregistry_output
+
+
+@pytest.mark.parametrize("registry_id", list(range(MAX_REGISTRIES)))
+@pytest.mark.parametrize("pool_index", list(range(MAX_POOLS_TO_CHECK)))
+def test_get_pool_params_stableswap_cryptoswap(
+    metaregistry, registries, max_pools, registry_id, pool_index, registry_pool_count
+):
+    registry, pool = get_registry_and_pool(
+        registries, registry_id, max_pools, pool_index, registry_pool_count
+    )
+
+    """This test is only for stableswap and cryptoswap amms"""
+
+    print(f"testing get_pool_params for pool: {pool}")
+    metaregistry_output = metaregistry.get_pool_params(pool)
+    print(f"metaregistry output: {metaregistry_output}")
+    actual_pool_params = [0] * 20
+
+    # A
+    if registry_id != METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+        actual_pool_params[0] = registry.get_A(pool)
+    else:
+        actual_pool_params[0] = curve_pool_v2(pool).A()
+
+    # D
+    if registry_id == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX:
+        actual_pool_params[1] = registry.get_D(pool)
+    elif registry_id == METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+        actual_pool_params[1] = curve_pool_v2(pool).D()
+
+    # gamma
+    if registry_id == METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX:
+        actual_pool_params[2] = registry.get_gamma(pool)
+    elif registry_id == METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
```

Comment:
these two tests (D/Gamma) can be merged so you only have 2 conditional statements instead of 4.

Reply:
Done.